txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/OutboundBridgeManager.java

```diff
@@ -66,18 +69,21 @@
 
         try
         {
-            Transaction transaction = (Transaction)TransactionManager.transactionManager().getTransaction();
+            Transaction transaction = (Transaction)TransactionManager.transactionManager(new InitialContext()).getTransaction();
 
-            Uid externalTxId = transaction.get_uid();
+            // generating new Uid to get it connected to incoming transaction
+            // the incoming transaction could come from different TM than Narayana is (e.g. Wildfly Transaction client)
+            transctionToUidMappings.putIfAbsent(transaction, new Uid());
```

Comment:
@tomjenkinson I think that usage of `Map` and getting records from it is based on `Transaction.equals()` implementation. I hope that current code relies on the `transaction.equal()`. Am I wrong here?